Cas 14113

nschweig-nrao opened the pull request:
Added fringe fit test, and merged changes from main back into the branch

amcnicho commented on benchmarks/bench_fringefit.py:
The longest running of the new tests are `Fringefit_corrcomb` and it took about 20 minutes total to complete a full run of the new tests in my sandbox.
```
· Running 10 total benchmarks (1 commits * 1 environments * 10 benchmarks)
[  0.00%] · For casa6 commit a943bb15 <6.5.6.17>:
[  0.00%] ·· Benchmarking virtualenv-py3.8-pip+casatasks6.5.6.17-pip+casatestutils6.5.6.17
[  3.33%] ··· Running (bench_fringefit.FreqMetaTests.time_metadata--)..
[ 10.00%] ··· Running (bench_fringefit.Fringefit_corrcomb.time_comb_none--).
[ 13.33%] ··· Running (bench_fringefit.Fringefit_dispersive_tests.time_manual_phase_cal--)..
[ 20.00%] ··· Running (bench_fringefit.Fringefit_refant_bookkeeping_tests.time_bookkeeping--)..
[ 26.67%] ··· Running (bench_fringefit.Fringefit_single_tests.time_single--)...
[ 36.67%] ··· Running (bench_fringefit.FreqMetaTests.time_metadata--)..
[ 43.33%] ··· Running (bench_fringefit.Fringefit_corrcomb.time_comb_none--).
[ 46.67%] ··· Running (bench_fringefit.Fringefit_dispersive_tests.time_manual_phase_cal--)..
[ 53.33%] ··· Running (bench_fringefit.Fringefit_refant_bookkeeping_tests.time_bookkeeping--)..
[ 60.00%] ··· Running (bench_fringefit.Fringefit_single_tests.time_single--)...
[ 70.00%] ··· bench_fringefit.FreqMetaTests.time_metadata                                                                  1.15±0s
[ 73.33%] ··· bench_fringefit.Fringefit_corrcomb.time_comb_all                                                          59.1±0.05s
[ 76.67%] ··· bench_fringefit.Fringefit_corrcomb.time_comb_none                                                         23.3±0.03s
[ 80.00%] ··· bench_fringefit.Fringefit_dispersive_tests.time_manual_phase_cal                                          2.08±0.01s
[ 83.33%] ··· bench_fringefit.Fringefit_mbd.time_mbd                                                                    1.70±0.04s
[ 86.67%] ··· bench_fringefit.Fringefit_refant_bookkeeping_tests.time_bookkeeping                                       1.40±0.02s
[ 90.00%] ··· bench_fringefit.Fringefit_single_tests.time_param                                                         13.0±0.05s
[ 93.33%] ··· bench_fringefit.Fringefit_single_tests.time_single                                                        1.59±0.02s
[ 96.67%] ··· bench_fringefit.Fringefit_tests.time_sbd                                                                  3.67±0.01s
[100.00%] ··· bench_fringefit.Fringefit_tests.time_uvrange                                                              7.12±0.02s

real    21m3.841s
user    19m26.630s
sys     0m21.472s
```

amcnicho reviewed (commented):
Looks good overall thanks for adding the new tests @nschweig-nrao.

smcastro commented on benchmarks/bench_fringefit.py:
```diff
@@ -0,0 +1,213 @@
+min_run_count = 5     # enforce the min_repeat * rounds setting is met
+timeout = 3600        # conservative 1hr hard cap for duration of a single test execution
+
+datapath = ctsys.resolve('unittest/fringefit/')
```
Please rename the class to a name without the task name as the file already have it. This is to make the appearance of names in the ASV menu page shorter and more descriptive. For example call it:
class SingleBandDelay() using Camel case style. Add any test case from the original script that uses sbd cal tables to this class. Do the same for the next classes and try to group together common test cases.

smcastro commented on benchmarks/bench_fringefit.py:
```diff
@@ -0,0 +1,213 @@
+        shutil.rmtree(self.uvfile, True)
+        shutil.rmtree('uvrange_with.cal', True)
+
+    def time_sbd(self):
```
Please add a doc string and refer to the name of the original test case in the string such as we have done for bench_flagdata test cases.

nschweig-nrao replied:
I've added those changes now. I also removed one test case from the CorrComb class. The test was using corrcomb='none' which will skip the interesting functionality of the parameter, and was only being used as a basis for comparison in the original test.

smcastro commented:
@amcnicho , @Jan-Willem . The changes look good to me. If the tests are passing locally, from my side they are good to go. Thank you Neal for applying the changes.

nschweig-nrao commented:
@smcastro somewhere in the cleanup for imports I had removed ctsys which was needed for finding the data path. I've fixed that now and everything appears to look good. Unless anyone has additional suggestions I can go ahead with the merge.
